### PR TITLE
Add announcement wrap to department radio messages for non-player sources (salvage magnet)

### PR DESCRIPTION
**Pull request by HoofedEar**
## About the PR 
The salvage magnet no longer "says" anything. This PR provides a way to define a radio message as an announcement, so that the message is a lot cleaner looking over the radio.
This also accounts for the requested change in #10001 (if-else)

**Screenshots**

image

**Changelog**

:cl:
- tweak: Salvage magnet can no longer speak

### CODE REVIEW

**Comment by Just-a-Unity-Dev** [comment]
Maybe instead of flat out removing the "says", how about use "beeps" instead?

**Comment by HoofedEar** [comment]
> Maybe instead of flat out removing the "says", how about use "beeps" instead?

Even with "beeps" it still looks a bit messy since it would say "Salvage magnet beeps:" so it would be _named_ Salvage magnet which is very odd and confusing. Maybe if the magnet was possessed by a player? But this is just to non-sentient interactions

File: Content.Server/Headset/HeadsetComponent.cs [hunk 1]
```diff
@@ -52,11 +52,16 @@ public bool CanListen(string message, EntityUid source, RadioChannelPrototype? p
             return prototype != null && Channels.Contains(prototype.ID) && RadioRequested;
         }
 
-        public void Receive(string message, RadioChannelPrototype channel, EntityUid source)
+        public void Receive(string message, RadioChannelPrototype channel, EntityUid source, bool announcement = false)
         {
-            if (!Channels.Contains(channel.ID) || !Owner.TryGetContainer(out var container)) return;
+            if (!Channels.Contains(channel.ID) || !Owner.TryGetContainerMan(out var containerMan))
```
Comment (mirrorcult) [inline Content.Server/Headset/HeadsetComponent.cs:57]: use `ContainerSystem` methods for this thru `_entMan.System<ContainerSystem>`

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by ElectroJr** [comment]
I'm currently scrapping IListen and IRadio, which will probably conflict with this pretty heavily.

**Comment by HoofedEar** [comment]
Marked as draft until #11980 is complete. Changes should be easy enough

**Comment by HoofedEar** [comment]
Gonna remake this after Crimbo

**Pull request closed without merging**